### PR TITLE
LPS 134949 invocation context

### PR DESCRIPTION
**Pull request by JorgeFerrer**
- LPS-134949 New interface to declare providers of invocation context objects
- LPS-134949 Interfaces for common invocation context objects
- LPS-134949 New module to host implementations of context providers
- LPS-134949 Initial implementations of context providers for Company, Group and user based on ThreadLocal

### CODE REVIEW

File: modules/apps/portal/portal-context/src/main/java/com/liferay/portal/context.internal/CompanyInvocationContextProviderImpl.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+@Component(service = CompanyInvocationContextProvider.class)
+public class CompanyInvocationContextProviderImpl
+	implements CompanyInvocationContextProvider {
+
+	@Override
+	public Company getCurrent() {
+		try {
+
+			// TODO: What is the best way to handle when the current invocation
```
Comment (Preston-Crary) [inline modules/apps/portal/portal-context/src/main/java/com/liferay/portal/context.internal/CompanyInvocationContextProviderImpl.java:38]: I would use fetchComany() and return null.

Comment (Preston-Crary) [reply modules/apps/portal/portal-context/src/main/java/com/liferay/portal/context.internal/CompanyInvocationContextProviderImpl.java:38]: Same with Group.

Comment (JorgeFerrer) [reply modules/apps/portal/portal-context/src/main/java/com/liferay/portal/context.internal/CompanyInvocationContextProviderImpl.java:38]: Makes sense. 

Thanks for the suggestion.

Comment (JorgeFerrer) [reply modules/apps/portal/portal-context/src/main/java/com/liferay/portal/context.internal/CompanyInvocationContextProviderImpl.java]: I've added a commit introducing a method called `isPresent()` to allow checking if an object is available in the current invocation context.

**Comment by JorgeFerrer** [comment]
Latest version at https://github.com/JorgeFerrer/liferay-portal/pull/912

**Pull request closed without merging**